src/MSBuild.Abstractions/MsBuildExtensions.cs

```diff
@@ -16,30 +16,37 @@ static object GetPropertyValue(object obj, string propertyName)
             {
                 if (obj == null)
                 {
-                    throw new ArgumentNullException("obj");
+                    throw new ArgumentNullException(nameof(obj));
                 }
 
                 var objType = obj.GetType();
                 var propInfo = GetPropertyInfo(objType, propertyName);
                 if (propInfo == null)
                 {
-                    throw new ArgumentOutOfRangeException("propertyName",
+                    throw new ArgumentOutOfRangeException(nameof(propertyName),
                       string.Format("Couldn't find property {0} in type {1}", propertyName, objType.FullName));
                 }
 
-                return propInfo.GetValue(obj, null);
+                var propertyValue = propInfo.GetValue(obj, null);
+                if (propertyValue == null)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(propertyName),
+                      string.Format("Null value for property {0} in type {1}", propertyName, objType.FullName));
+                }
+                return propertyValue;
             }
 
-            static PropertyInfo GetPropertyInfo(Type type, string propertyName)
+            static PropertyInfo? GetPropertyInfo(Type type, string propertyName)
             {
-                PropertyInfo propInfo;
+                PropertyInfo? propInfo;
```

Comment:
>propInfo [](start = 30, length = 8)

I don't think the intention is that this ever returns null. perhaps throw and exception in that case? I would rather have an exception for this than needing to track it down through other means

Reply:
I have moved the null check to within `GetPropertyInfo` and marked it as not nullable.